components/Tabs/src/private/MDCItemBar.m

```diff
@@ -489,6 +490,9 @@ - (void)stopObservingItems {
 }
 
 - (UIUserInterfaceSizeClass)horizontalSizeClass {
+  if (self.tabBar && self.tabBar.sizeClassDelegate && [self.tabBar.sizeClassDelegate respondsToSelector:@selector(horizontalSizeClassForObject:)]) {
```

Comment:
I'd consider using a local variable to hold a reference to the delegate to avoid TOCTOU issues. I also think it might simplify readability: http://go/gh/stephanemoore/archives/blob/master/objc/pitfalls/tocttou.md

```
id<MDCTabBarSizeClassDelegate> sizeClassDelegate = self.tabBar.sizeClassDelegate;
if ([sizeClassDelegate respondsToSelector:@selector(horizontalSizeClassForObject:)]) {
  return [sizeClassDelegate horizontalSizeClassForObject:self];
}
```

Reply:
Thanks, that seems nearly impossible for single-thread code like this (the linked doc didn't explain how single-threading could have these race conditions),  I'll update with an automatic variable.